core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -509,6 +511,13 @@ private[spark] class ExecutorAllocationManager(
   private def onExecutorBusy(executorId: String): Unit = synchronized {
     logDebug(s"Clearing idle timer for $executorId because it is now running a task")
     removeTimes.remove(executorId)
+
+    // Executor is added to remove by misjudgment due to async listener making it as idle).
+    // see SPARK-9552
+    if (executorsPendingToRemove.contains(executorId)) {
```

Comment:
Yes. I know that. From the API design and implementation (named as `killExecutors`),  I'd prefer more general case. In case someone else calls that in the future. Besides, batch kill is better than kill them one by one each time. If it is ok not to take that account, I will handle that according to existing case. 

Thanks for your comments and suggestions. I will change the code accordingly.